uart_emu/include/rmt_uart.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "esp_err.h"
+#include "freertos/FreeRTOS.h"
+#include "soc/soc_caps.h"
+#include "hal/gpio_types.h"
+#include "rmt_uart_type.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+    uint32_t baud_rate;                   /*!< UART baud rate*/
+    size_t tx_trans_queue_depth;          /*!< UART tx internal transfer queue depth, increase this value can support more transfers pending in the background */
+    size_t rx_buffer_size;                /*!< UART rx fifo size*/
+    uart_emu_word_length_t data_bits;     /*!< UART byte size*/
+    uart_emu_parity_t parity;             /*!< UART parity mode*/
+    uart_emu_stop_bits_t stop_bits;       /*!< UART stop bits*/
+    gpio_num_t tx_io_num;                 /*!< UART tx io number*/
+    gpio_num_t rx_io_num;                 /*!< UART rx io number*/
+    struct {
+        uint32_t with_dma: 1;             /*!< If set, the driver will allocate an RMT channel with DMA capability */
+    } flags;                              /*!< UART config flags */
+} rmt_uart_config_t;
+
+/**
+ * @brief Structure for defining callback functions for RMT UART transmission events.
+ */
+typedef struct {
+    rmt_uart_tx_done_callback_t on_tx_trans_done;           /*!< Callback function for handling the completion of a transmission */
+} rmt_uart_event_tx_callbacks_t;
+
+/**
+ * @brief Structure for defining callback functions for RMT UART receive events.
+ */
+typedef struct {
+    rmt_uart_rx_done_callback_t on_rx_trans_done;           /*!< Callback function for handling the completion of a reception */
+} rmt_uart_event_rx_callbacks_t;
+
+/**
+ * @brief Create a new RMT UART controller.
+ *
+ * @param[in] uart_config Configuration for the RMT UART controller.
+ * @param[out] ret_device Handle to the RMT UART controller.
+ *
+ * @return
+ * - `ESP_OK`: RMT UART controller was successfully created.
+ * - `ESP_ERR_INVALID_ARG`: Invalid arguments (e.g., null `uart_config` pointer or `ret_device` pointer).
+ * - `ESP_ERR_NO_MEM`: No enough memory to create the RMT UART controller.
+ */
+esp_err_t rmt_new_uart_device(const rmt_uart_config_t *uart_config, rmt_uart_device_handle_t *ret_device);
```

Comment:
I hope the APIs in this "uart_emu" component could share the same name prefix "uart_emu". 

e.g. `uart_emu_new_from_rmt`

Reply:
if an API is RMT specific, we can put it in uart_emu_rmt.h, otherwise, we can put it in uart_emu.h